File: backtracking/combinations.py
class Solution:
    def combine(self, n: int, k: int) -> List[List[int]]:
        result = []  # Store results set
        self.backtracking(n, k, 1, [], result)
        return result
    def backtracking(self, n, k, startIndex, path, result):
        if len(path) == k:
            result.append(path[:])
            return
        for i in range(startIndex, n - (k - len(path)) + 2):  # Optimized section
            path.append(i)  # Handle node
            self.backtracking(n, k, i + 1, path, result)
            path.pop()  # Backtrack and undo node handling

File: backtracking/ts/combinations.ts
function combine(n: number,k: number): number[][] {
    let resArr: number[][] = [];
    function backTracking(n: number,k: number, startIndex: number,tempArr: number[]): void {
        if (tempArr.length === k) {
            resArr.push(tempArr.slice());
            return;
        }


        for (let i = startIndex;i <= n -k + tempArr.length;i++) {
            tempArr.push(i);
            backTracking(n,k,i+1,tempArr);
            tempArr.pop();
        }        
    }
      backTracking(n,k,1,[]);
    
    return resArr
}

